spec/order_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'order'
+
+# potential for more functionality - adding several of one item at once.
+
+describe Order do
+  let(:twilio_adapter) { double(:twilio_adapter, :send_message => "sends a message") }
+  let(:dish1) { double(:dish, :name => "Baked Potato", :price => 3.25) }
+  let(:dish2) { double(:dish, :name => "Spaghetti Bolognese", :price => 4.70) }
+  let(:test_menu) { double(:menu, :title => "Fake Ass Restaurant", :dishes => [dish1, dish2], :pick => dish1) }
+  let(:test_order) { described_class.new(test_menu, twilio_adapter, '7777777777777') }
+
+  describe '#initialize' do
+    it 'has menu stored to menu attribute on initialization' do
+      expect(test_order.menu).to eq test_menu
+    end
+    it 'has balance of zero' do
+      expect(test_order.total).to match(/^£0.00$/)
+    end
+    it 'has empty list of dishes' do
+      expect(test_order.dishes).to eq []
+    end
+    it 'is set to incomplete' do
+      expect(test_order.complete).to be false
+    end
+  end
+
+  describe '#add_items' do
+    it 'adds a dish to the dishes array' do
+      expect { test_order.add_items(dish1) }.to change { test_order.dishes.length }.by 1
+    end
+    it 'adds multiple items at once, and multiple of the same' do
+      expect { test_order.add_items(dish1, dish1, dish1, dish1) }.to change { test_order.dishes.length }.by 4
+    end
+
+    it 'throws an error if the order has already been finalized' do
+      test_order.finalize
+      expect { test_order.add_items(dish2) }.to raise_error "this order is closed"
+    end
+    it 'adds the price of the added items to the balance' do
+      expect { test_order.add_items(dish1, dish1) }.to change { test_order.balance }.by(dish1.price * 2)
+    end
+
```

Comment:
overall really like the readability of your tests, it's easy to follow and see what they are doing